integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/AlterTableTestCase.scala

```diff
@@ -1022,7 +1022,31 @@ class AlterTableTestCase extends QueryTest with BeforeAndAfterAll {
       assert(exception.getMessage.contains("Unsupported alter operation on hive table"))
     } else if (SparkUtil.isSparkVersionXandAbove("2.2")) {
       sql("alter table alter_hive add columns(add string)")
-      sql("insert into alter_hive select 'abc','banglore'")
+      sql("alter table alter_hive add columns (var map<string, string>)")
+      sql("insert into alter_hive select 'abc','banglore',map('age','10','birth','2020')")
+      checkAnswer(
+        sql("select * from alter_hive"),
+        Seq(Row("abc", "banglore", Map("age" -> "10", "birth" -> "2020")))
+      )
+    }
+  }
+
+  test("Alter table add column for hive partitioned table for spark version above 2.1") {
+    sql("drop table if exists alter_hive")
+    sql("create table alter_hive(name string) stored as rcfile partitioned by (dt string)")
+    if (SparkUtil.isSparkVersionXandAbove("2.2")) {
+      sql("alter table alter_hive add columns(add string)")
+      sql("alter table alter_hive add columns (var map<string, string>)")
```

Comment:
Can you please add a case with STRUCT Datatype also

Reply:
It does not work with default STRUCT Datatype (struct<>)

Reply:
why it doesn't work with Struct type?

Reply:
`DataTypeUtil.valueOf` just return default struct type (struct<>), here should work more for this.

Reply:
If this PR is just to support map, please modify the PR title and description.
To support struct type, need another PR?

Reply:
Yes, this PR only support `map<string, string>, array<string>, struct<>` currently. I can do some work to fully support this.

Reply:
Ok, I will merge this PR. Please raise another JIRA ticket for the struct support. Thanks